cdap-app-templates/cdap-etl/hydrator-spark-core/src/main/java/co/cask/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -55,10 +48,18 @@
 /**
  * Base Spark program to run a Hydrator pipeline.
  */
-public abstract class SparkPipelineDriver {
+public abstract class SparkPipelineRunner {
 
-  protected abstract SparkCollection<Object> getSource(String stageName,
-                                                       PluginFunctionContext pluginFunctionContext) throws Exception;
+  protected abstract SparkCollection<Object> getSource(StageInfo stageInfo) throws Exception;
+
+
+  protected abstract SparkPairCollection<Object, Object> addJoinKey(
+    StageInfo stageInfo, String inputStageName,
```

Comment:
I can't, the last argument will break checkstyle because the line will be too long.